src/script/lua_api/l_mainmenu.cpp

```diff
@@ -672,7 +672,8 @@ int ModApiMainMenu::l_delete_world(lua_State *L)
 
 		std::vector<std::string> paths;
 		paths.push_back(spec.path);
-		fs::GetRecursiveSubPaths(spec.path, paths);
+		std::vector<char> ignore;
+		fs::GetRecursiveSubPaths(spec.path, paths, true, ignore);
```

Comment:
This modification shouldn't be necessary if you designed the modification to GetRecursiveSubPaths better.